docs/content/en/docs/user-guide/event-watcher.md

```diff
@@ -0,0 +1,152 @@
+description: >
+  A helper facility to automatically update files when it finds out a new event.
+---
+
+The only way to upgrade your application with PipeCD is modifying configuration files managed by the Git repositories.
+It brings benefits quite a bit, but it can be painful to manually update them every time in some cases (e.g. continuous deployment to your development environment for debugging, the latest prerelease to the staging environment).
+
+If you're experiencing any of the above pains, Event watcher is for you.
+Event watcher works as a helper facility to seamlessly link CI and CD. This feature lets you automatically update files managed by your Piped when an arbitrary event occurs.
+While it empowers you to build pretty versatile workflows, the canonical use case is that you trigger a new deployment by image updates, package releases, etc.
+
+This guide walks you through configuring Event watcher and how to push an Event.
+
+## Prerequisites
+Before configuring EventWatcher, be sure to grant write permission to the SSH key used by Piped. See [here](/docs/operator-manual/piped/configuring-event-watcher/) for more details.
+
+## Usage
+File updating can be done by registering the latest value corresponding to the Event in the control-plane and comparing it with the current value.
+
+Therefore, you mainly need to:
+1. define which values in which files should be updated when a new Event found.
+1. integrate a step to push an Event to the control-plane using `pipectl` into your CI workflow.
+
+### 1. Define Events
+Prepare EventWatcher files placed at the `.pipe/` directory at the root of the Git repository.
+In that files, you define which values in which files should be updated when the Piped found out a new Event.
+
+For instance, suppose you want to update the Kubernetes manifest defined in `helloworld/deployment.yaml` when an Event with the name `helloworld-image-update` occurs:
+
+```yaml
+apiVersion: pipecd.dev/v1beta1
+kind: EventWatcher
+spec:
+  events:
+    - name: helloworld-image-update
+      replacements:
+        - file: helloworld/deployment.yaml
+          yamlField: $.spec.template.spec.containers[0].image
+```
+
+The full list of configurable `EventWatcher` fields are [here](/docs/user-guide/configuration-reference/#event-watcher-configuration).
+
+### 2. Pushing an Event with `pipectl`
+To register a new value corresponding to Event such as the above in the control-plane, you need to perform `pipectl`.
+And if you want to push Events continuously, we highly recommend integrating a step for that into your CI workflow.
+
+According to [this guide](http://localhost:1313/docs/user-guide/command-line-tool/#installation), you first install it on your CI system or where you want to run.
+And then you grab the API key for the control-plane according to [this guide](http://localhost:1313/docs/user-guide/command-line-tool/#authentication).
+
+Once you're all set up, now all we need to do is to execute a command like:
```

Comment:
"Once you're all set up, pushing a new Event to PipeCD by the following command:"